resources/ts/plan-details.ts

```diff
@@ -0,0 +1,199 @@
+        method: 'POST',
+        headers: {
+            'Content-Type' : 'application/json',
+            'X-CSRF-Token': csrfToken,
+        },
+    })
+    .then(response => response.json())
+    .then(data => {
+        let otherPlaceHtml: string = '';
+        let addPlaceHtml: string = '';
+        data['data'].forEach(((place: { [key: string]: string }, index: number) => {
+            if (index === 0 || (index + 1) % 4 === 0) {
+                otherPlaceHtml += '<div class="row mb-3">'
+                addPlaceHtml +=  '<div class="row mb-3">'
+            }
+
+            otherPlaceHtml += `<div class="col-4" id="other_place_${index}">
+            <div>
+                <img src="${data['asset_path']}/${place['image']}" class="img-sm" alt="${data['asset_path']}/${place['image']}">
+            </div>
+                <p data-place-id="${place['id']}">${place['name_en']}</p>
+            </div>`
+            if ((index + 1) % 3 === 0) {
+                otherPlaceHtml += `</div>`;
+            }
+
+            addPlaceHtml += `<div class="col-4 add-place d-flex flex-column" id="add_place_${index}">
+            <div class="mx-auto">
+                <img src="${data['asset_path']}/${place['image']}" class="img-sm" alt="${data['asset_path']}/${place['image']}">
+            </div>
+                <p class="mt-auto" data-place-id="${place['id']}">${place['name_en']}</p>
+            </div>`
+
+            if ((index + 1) % 3 === 0) {
+                otherPlaceHtml += `</div>`;
+                addPlaceHtml += `</div>`;
+            }
+        });
+
+        const otherPlaceElement: JQuery<HTMLElement> = $('#other_place_list');
+        otherPlaceElement.empty().append(otherPlaceHtml);
+
+        const addPlaceElement: JQuery<HTMLElement> = $('#add_place_list');
+        addPlaceElement.empty().append(addPlaceHtml);
+
+        const placeList = document.querySelectorAll(".add-place") as NodeListOf<HTMLElement>;
+        let selectedElement: HTMLElement | null = null;
+
+        function clickAddPlace(placeList: HTMLElement) {
+            const addPlaceButton = document.getElementById("add_place_button") as HTMLButtonElement;
+            addPlaceButton.disabled = false;
+        
+            if (selectedElement) {
+                selectedElement.style.border = "";
+                selectedElement.style.backgroundColor = "";
+                selectedElement.classList.remove("selected-place");
+            }
+        
+            placeList.style.border = "2px solid #C0C0C0";
+            placeList.style.backgroundColor = "#F3F3F3";
+            placeList.classList.add("selected-place");
+            selectedElement = placeList;
+        }
+        
+        placeList.forEach((placeList) => {
+            placeList.addEventListener("click", () => {
+                clickAddPlace(placeList);
+            });
+        });
+    })
+    .catch(error => {
+        console.error('Error', error);
+    });
```

Comment:
Please remove console logs. Maybe it is better to create a new element that shows "Error, something went wrong" so that the user can know if the fetch eventually failed. But since I advised to use await, surround this code with try catch.